### PR TITLE
fix: datasource specs cannot be loaded during collection

### LINKED ISSUE #3866: "datasource" specs cannot be used as depend component correctly
When making a new datasource spec, that depends on another `existing datasource` specs, it always reports `Missing Requirment` due to the `content` of the existing datasource was cleared as `None`, see the following debug information:

```
2023-08-01 17:16:17,036     INFO insights.core.dr Trying insights.specs.default.DefaultSpecs.intersystems_iris_cpf
-------------------
insights.core.spec_factory.DatasourceProvider.__init__
obj_id= 140573762980216
obj= <DatasourceProvider(path='/tmp/iris.cpf', cmd=''>
obj.content= ['[ConfigFile]', 'Product=IRIS', 'Version=2023.1', '', '[Databases]', 'IRISSYS=/tmp/', 'IRISLIB=/intersystems/mgr/irislib/', 'IRISTEMP=/intersystems/mgr/iristemp/', 'IRISLOCALDATA=/intersystems/mgr/irislocaldata/', 'IRISAUDIT=/intersystems/mgr/irisaudit/', 'ENSLIB=/intersystems/mgr/e>
2023-08-01 17:16:17,036     INFO insights.core.dr Trying insights.specs.Specs.intersystems_iris_cpf
2023-08-01 17:16:17,037     INFO insights.core.dr Trying insights.parsers.iris.IrisCpf

-------------------
insights.core.plugins.parser.invoke
obj_id= 140573762980216
obj= <DatasourceProvider(path='/tmp/iris.cpf', cmd=''>
obj.content= None
2023-08-01 17:16:17,039     INFO insights.core.dr Trying insights.specs.default.DefaultSpecs.intersystems_iris_messages_log
2023-08-01 17:16:17,039    DEBUG insights.core.dr insights.specs.default.DefaultSpecs.intersystems_iris_messages_log missing requirements All: ['insights.parsers.iris.IrisCpf'] Any:
```

### PR DESCRIPTION
**Pull request by xiangce**
- fix #3866
- and fix the infinite loop between load() and content() of DatasourceProvider

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### CODE REVIEW

**Comment by xiangce** [comment]
@ryan-blakley, @chenlizhong - I found this problem when trying to refine PR#3861 raised by @wushiqinlou. It makes sense to me to take an existing datasource spec as a dependency of a new spec. What do you think? Please kindly review.

File: insights/core/spec_factory.py [hunk 1]
```diff
@@ -80,7 +80,8 @@ def content(self):
 
         if self._content is None:
             try:
-                self._content = self.load()
+                if not self.loaded:
```
Comment (ryan-blakley) [inline insights/core/spec_factory.py:83]: This is a bit of a nitpick, but could we add this check to the if statement on line 81 above?

Comment (xiangce) [reply insights/core/spec_factory.py]: Makes sense, modified. Please check again

**Comment by xiangce** [comment]
test me

**Comment by xiangce** [comment]
> I can't remember why we set `._content` to `None` after the datasource is written other than maybe trying to save memory. From the original commit it looks like we were working on support for openshift collection and might not have considered dependencies. In any case, it seems like a bug, and the fix lgtm.

@csams, Thank you so much for the midnight review comment! Have a good night :-)

**Comment by xiangce** [comment]
test me

**Pull request merged**